pkg/scheduler/plugins/elasticquota/core/quota_info.go

```diff
@@ -0,0 +1,192 @@
+
+package core
+
+import (
+	"sync"
+
+	v1 "k8s.io/api/core/v1"
+	quotav1 "k8s.io/apiserver/pkg/quota/v1"
+	"sigs.k8s.io/scheduler-plugins/pkg/apis/scheduling/v1alpha1"
+)
+
+type QuotaCalculateInfo struct {
+	Max                       v1.ResourceList `json:"max"`
+	OriginalMin               v1.ResourceList `json:"original_min"`
+	AutoScaleMin              v1.ResourceList `json:"auto_scale_min"`
+	Used                      v1.ResourceList `json:"used"`
+	Request                   v1.ResourceList `json:"request"`
+	SharedWeight              v1.ResourceList `json:"shared_weight"`
+	Runtime                   v1.ResourceList `json:"runtime"`
+	SubGroupsTotalResLastTime v1.ResourceList `json:"sub_groups_total_res_last_time"`
+}
+
+type QuotaInfo struct {
+	Name              QuotaNameType      `json:"name"`
+	ParentName        QuotaNameType      `json:"parent_name"`
+	IsParent          bool               `json:"is_parent"`
+	RuntimeVersion    int64              `json:"runtimeVersion,omitempty"`
+	AllowLentResource bool               `json:"min_quota_share"`
+	CalculateInfo     QuotaCalculateInfo `json:"calculateInfo,omitempty"`
+	lock              sync.Mutex
+}
+
+func NewEmptyQuotaInfo(name QuotaNameType) *QuotaInfo {
+	return &QuotaInfo{
+		Name: name,
+	}
+}
+
+func NewQuotaInfo(isParent, allowLentResource bool, name, parentName QuotaNameType) *QuotaInfo {
+	return &QuotaInfo{
+		Name:              name,
+		ParentName:        parentName,
+		IsParent:          isParent,
+		AllowLentResource: allowLentResource,
+		RuntimeVersion:    0,
+		CalculateInfo: QuotaCalculateInfo{
+			Max:                       v1.ResourceList{},
+			AutoScaleMin:              v1.ResourceList{},
+			OriginalMin:               v1.ResourceList{},
+			Used:                      v1.ResourceList{},
+			Request:                   v1.ResourceList{},
+			SharedWeight:              v1.ResourceList{},
+			Runtime:                   v1.ResourceList{},
+			SubGroupsTotalResLastTime: v1.ResourceList{},
+		},
+	}
+}
+
+func (qi *QuotaInfo) LockDeepCopy() *QuotaInfo {
+	qi.lock.Lock()
+	defer qi.lock.Unlock()
+
+	return &QuotaInfo{
+		Name:              qi.Name,
+		ParentName:        qi.ParentName,
+		IsParent:          qi.IsParent,
+		AllowLentResource: qi.AllowLentResource,
+		RuntimeVersion:    qi.RuntimeVersion,
+		CalculateInfo: QuotaCalculateInfo{
+			Max:                       qi.CalculateInfo.Max.DeepCopy(),
+			AutoScaleMin:              qi.CalculateInfo.AutoScaleMin.DeepCopy(),
+			OriginalMin:               qi.CalculateInfo.OriginalMin.DeepCopy(),
+			Used:                      qi.CalculateInfo.Used.DeepCopy(),
+			Request:                   qi.CalculateInfo.Request.DeepCopy(),
+			SharedWeight:              qi.CalculateInfo.SharedWeight.DeepCopy(),
+			Runtime:                   qi.CalculateInfo.Runtime.DeepCopy(),
+			SubGroupsTotalResLastTime: qi.CalculateInfo.SubGroupsTotalResLastTime.DeepCopy(),
+		},
+	}
+}
+
+func (qi *QuotaInfo) LockFillQuotaInfoFrom(rhs *QuotaInfo) {
```

Comment:
"LockFillQuotaInfoFrom" function is strange,  there are so many variables in QuotaInfo, and the name "LockFillQuotaInfoFrom" looks like want to copy everything, but it doesn't.